libraries/bootstrap.php

```diff
@@ -41,16 +41,17 @@ class_exists('\\Joomla\\CMS\\Autoload\\ClassLoader');
 // Decorate Composer autoloader
 spl_autoload_register([new \Joomla\CMS\Autoload\ClassLoader($loader), 'loadClass'], true, true);
 
-// Check if load of the environment variables is enabled.
-if (is_file(JPATH_ROOT . '/.env') || is_file(JPATH_ROOT . '/.env.dev')) {
+// Checking whether loading of the environment variables from the file is enabled. Then parse them.
+if (is_file(JPATH_ROOT . '/.env')) {
     /**
      * Some PHP configurations set variables_order="GPCS" not "EGPCS", in which case $_ENV is NOT populated.
      * Detect if the $_ENV  was empty and handle it by explicitly populating through getenv() for CLI,
      * or filter variables from  $_SERVER prefixed with JOOMLA_ for web requests.
+     *
      * To make it work the same for CLI and WEB set variables_order="EGPCS" in php.ini.
      */
     if (!$_ENV) {
-        $_ENV = PHP_SAPI === 'cli' ? getenv() : array_filter($_SERVER, fn ($k) => str_starts_with($k, 'JOOMLA_'), ARRAY_FILTER_USE_KEY);
+        $_ENV = !\array_key_exists('REQUEST_METHOD', $_SERVER) ? getenv() : array_filter($_SERVER, fn ($k) => str_starts_with($k, 'JOOMLA_'), ARRAY_FILTER_USE_KEY);
```

Comment:
IMHO, the previous one was more clear and shorter.
```suggestion
        $_ENV = !isset($_SERVER['REQUEST_METHOD']) ? getenv() : array_filter($_SERVER, fn ($k) => str_starts_with($k, 'JOOMLA_'), ARRAY_FILTER_USE_KEY);
```

Reply:
Yes, right, but I changed to how other part of core code working.
In the end it probably can be just `$_ENV = getenv();`, need some time to check, I did not decided for now.

Reply:
> In the end it probably can be just $_ENV = getenv();

You should avoid using the `getenv()` because it can cause segmentation fault in some systems (e.g. Windows or Apache with threads).

I suggest this:

```php
$_ENV = $_SERVER ? array_filter($_SERVER, fn ($k) => str_starts_with($k, 'JOOMLA_'), ARRAY_FILTER_USE_KEY) : getenv();
```

Reply:
Or you can use the `$_SERVER` instead of `$_ENV`. Then all that code won't be needed.